### PR TITLE
Update tablegen.py to update the installation option table for 1.1

### PR DESCRIPTION
**Pull request by johnma14**
Updated tablegen.py to process the configuration options from the values.yaml
files under /istio/install/kubernetes/heml/subcharts directory and the
remaining configuration options like global, istiocoredns, istio_cni from
values.yaml under /istio/install/kubernetes/helm/istio directory.

### CODE REVIEW

File: scripts/tablegen.py [hunk 1]
```diff
@@ -37,13 +46,17 @@ def endOfTheList(context, lineNum, lastLineNum, totalNum):
     if lineNum != lastLineNum:
         return False, valueList
 
+
```
Comment (sdake) [inline scripts/tablegen.py]: extra whitespace

File: scripts/tablegen.py [hunk 2]
```diff
@@ -37,13 +46,17 @@ def endOfTheList(context, lineNum, lastLineNum, totalNum):
...
     for nextLineNum in range(lineNum + 1, totalNum):
         nextLine = context[nextLineNum]
+
         if  len(nextLine.lstrip()) != 0 and '#' != nextLine.lstrip()[0] and ':' in nextLine:
             if whitespaces >= (len(nextLine) - len(nextLine.lstrip())) / 2:
                 if flag == 0:
                     valueList.append(currentLine.split(':', 1)[1].strip())
                 return True, valueList
+            else:
```
Comment (sdake) [inline scripts/tablegen.py]: if this renders correctly, then the whitespace comparison is not needed and you should be able to delete line 54/58/59

File: scripts/tablegen.py [hunk 3]
```diff
@@ -58,73 +71,119 @@ def endOfTheList(context, lineNum, lastLineNum, totalNum):
...
+
+            storekey = key
+            sk = storekey.split('.', 2)
+            if len(sk) > 1:
+                storekey = '.'.join(sk[:1]).lstrip('.')
+            else:
+                storekey = '.'.join(sk[:0]).lstrip('.')
+
+            #
+            # If we are processing the configurations options within the values.yaml under istio,
```
Comment (sdake) [inline scripts/tablegen.py]: how does this condition occur?

File: scripts/tablegen.py (hunk 2 above)
Comment (sdake) [reply scripts/tablegen.py]: Note iirc, the reason for this whitespace comparison was to produce a description field - but my memory is fuzzy. @gyliu513 may know more.

**Comment by linsun** [comment]
@johnma14 great work!!!!   This is so well needed.

The rendered table looks good to me.  I agree with Steve's comment on parsing the comment for each field in the table, that would be really nice.

**Comment by johnma14** [comment]
@sdake @linsun thank you for your kind comments and review. Appreciate it. I will make those changes you recommended in this PR and I can take a look at filling the description fields also. Is there a WI associated to that already or should we create one?

File: scripts/tablegen.py (hunk 2 above)
Comment (johnma14) [reply scripts/tablegen.py]: @sdake without that if-else statement it doesn't render correctly. It only shows the content from the values.yaml under istio and not the values from the sub charts dir.

File: scripts/tablegen.py (hunk 3 above)
Comment (johnma14) [reply scripts/tablegen.py]: I have included comments in the latest PR. This condition is used to check if the configuration option is in values.yaml under istio directory and if this option is a new option or processed before (for example, if this was found in the sub charts directory). The assumption here is that the configuration options under the sub charts directory has the latest values and we want to keep them and only process the configuration options that are only in the istio/values.yaml dir like 'global' option for example. 

Hope this helps. Thanks for the review

File: scripts/tablegen.py (hunk 1 above)
Comment (johnma14) [reply scripts/tablegen.py]: fixed in the latest PR

File: scripts/tablegen.py (hunk 2 above)
Comment (sdake) [reply scripts/tablegen.py]: oh right that makes sense.

File: scripts/tablegen.py (hunk 3 above)
Comment (sdake) [reply scripts/tablegen.py]: The code currently (or did) work by replacing all contents between  `<!-- AUTO-GENERATED-START --` and `<!-- AUTO-GENERATED-END -->`.  We can't select out keys that are in the existing MD as the comments or default may have changed.  Can you take a look at ensuring this case works?  The code takes fractions of a second to run, so we aren't all that concerned about effeciency.

Comment (sdake) [reply scripts/tablegen.py]: > @sdake @linsun thank you for your kind comments and review. Appreciate it. I will make those changes you recommended in this PR and I can take a look at filling the description fields also. Is there a WI associated to that already or should we create one?

Can handle the description field as a separate PR. The existing issue for this is: https://github.com/istio/istio.io/issues/2566

Comment (johnma14) [reply scripts/tablegen.py]: @sdake This code works exactly like you mentioned, i.e, replace everything in the markdown file between .  That new if-else block processes the configuration options in the values.yaml under the istio/install/kubernetes/helm/istio directory. There are configuration options in there like:
- kiali : we want to make sure that the parameters for this option that will be populated in the index.md are from istio/install/kubernetes/helm/subcharts/kiali/values.yaml and not istio/install/kubernetes/helm/istio/values.yaml

- global: This does not have a corresponding directory under istio/install/kubernetes/helm/subcharts/ so we will get the parameters for this option from stio/install/kubernetes/helm/istio/values.yaml.

That was the thought process.

**Pull request merged**